compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,116 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+typedef enum RegisterStatus {
+   OK_FOR_REUSE,
+   DO_NOT_TOUCH
+} RegisterStatus;
+
+static inline RegisterStatus
+evaluateRegister(TR::Register **childReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+   RegisterStatus status;
+
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+   {
+      value = child->getLongInt();
+      if (constantIsUnsignedImm12(value))
+      {
+         //TODO: Cover special cases for immediate loads
+      }
+      else
+      {
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
+         status = OK_FOR_REUSE;
+      }
+	}
+	else
+   {
+      *childReg = cg->evaluate(child);
+      status = DO_NOT_TOUCH;
+	}
+   return status;
+   }
+
+static inline TR::Register *
+evaluateBooleanTypeInteger(TR::Node *node, TR::InstOpCode::Mnemonic regOp, TR::CodeGenerator *cg, uint8_t sizeOfOperation)
```

Comment:
I'm not sure I understand the name of this function.  What does "BooleanTypeInteger" mean?